### ISSUE TITLE
Message passing failure

### ISSUE THREAD

**Issue by CliffordAnderson** [NetsBlox/NetsBlox#2401]
When coding a simple chat client/server in class today, we experienced a strange JavaScript error. The messages stopped arriving, then came in a bunch. The server then repeatedly sent the same message over and again. A teal penumbra began to encircle the server code, slowly growing larger and larger. 

![chatroom](https://user-images.githubusercontent.com/1919466/54623004-aa761480-4a38-11e9-8721-660f759cc4be.png)

The problem was fixed by saving the clients and the server and having the entire class refresh and reload their projects.

**Comment by CliffordAnderson** [comment]
A guess is that a student might have copied the code for the server, creating an infinite messaging loop.

**Comment by hamidzr** [comment]
what we can see from the screenshot is that there are quite a few messages queued (~5k) but that shouldn't be an issue. 
did you try stopping the project? the surrounding teal must be a client issue. I can take a look at the logs to get some more insight. I can find it easier if I have a datetime and/or username. Could you email those to me please?

**Comment by CliffordAnderson** [comment]
Yes, we stopped the project and restarted to restore the functionality. My user name is andersoncliffb and we were working on the project ~10:30 a.m. on Tuesday, March 19th. Thanks for checking into the issue.

**Comment by hamidzr** [comment]
Was the chat server project running under your username?

**Comment by hamidzr** [comment]
Unfortunately, I couldn't find anything useful If this happens again please use the built-in bug report feature (click on the netsblox logo in the editor). That will try to capture the state of the project and should help us pinpoint the issue

**Comment by CliffordAnderson** [comment]
OK, good suggestion. If I see this happen again, I'll report the bug using the built-in mechanism.
